drivers/gpio/gpio_mchp_xec.c

```diff
@@ -329,260 +309,278 @@ static const struct gpio_driver_api gpio_xec_driver_api = {
 	.disable_callback = gpio_xec_disable_callback,
 };
 
-#ifdef CONFIG_GPIO_XEC_GPIO000_036
-static int gpio_xec_port000_036_init(struct device *dev);
+#ifdef DT_INST_0_MICROCHIP_XEC_GPIO
+
+DEVICE_DECLARE(gpio_xec_port0);
 
-static const struct gpio_xec_config gpio_xec_port000_036_config = {
+static struct gpio_xec_data gpio_xec_port0_data;
+
+static const struct gpio_xec_config gpio_xec_port0_config = {
 	.common = {
-		.port_pin_mask = GPIO_PORT_PIN_MASK_FROM_NGPIOS(DT_INST_0_MICROCHIP_XEC_GPIO_NGPIOS),
+		.port_pin_mask = DT_INST_0_MICROCHIP_XEC_GPIO_GPIO_PIN_MASK,
 	},
-	.pcr1_base = (u32_t *)  DT_GPIO_XEC_GPIO000_036_BASE_ADDR,
-	.port_num = MCHP_GPIO_000_036,
-#ifdef DT_GPIO_XEC_GPIO000_036_IRQ
-	.girq_id = MCHP_GIRQ11_ID,
+	.pcr1_base = (u32_t *) DT_INST_0_MICROCHIP_XEC_GPIO_BASE_ADDRESS_0,
+	.input_base = (u32_t *) DT_INST_0_MICROCHIP_XEC_GPIO_BASE_ADDRESS_1,
+	.output_base = (u32_t *) DT_INST_0_MICROCHIP_XEC_GPIO_BASE_ADDRESS_2,
+#ifdef DT_INST_0_MICROCHIP_XEC_GPIO_GIRQ
+	.girq_id = DT_INST_0_MICROCHIP_XEC_GPIO_GIRQ,
 	.flags = GPIO_INT_ENABLE,
 #else
 	.flags = 0,
 #endif
 };
 
-static struct gpio_xec_data gpio_xec_port000_036_data;
-
-DEVICE_AND_API_INIT(gpio_xec_port000_036, DT_GPIO_XEC_GPIO000_036_LABEL,
-		gpio_xec_port000_036_init,
-		&gpio_xec_port000_036_data, &gpio_xec_port000_036_config,
-		POST_KERNEL, CONFIG_KERNEL_INIT_PRIORITY_DEFAULT,
-		&gpio_xec_driver_api);
-
-static int gpio_xec_port000_036_init(struct device *dev)
+static int gpio_xec_port0_init(struct device *dev)
 {
-#ifdef DT_GPIO_XEC_GPIO000_036_IRQ
+#ifdef DT_INST_0_MICROCHIP_XEC_GPIO_GIRQ
 	const struct gpio_xec_config *config = dev->config->config_info;
 
 	/* Turn on the block enable in the EC aggregator */
 	MCHP_GIRQ_BLK_SETEN(config->girq_id);
 
-	IRQ_CONNECT(DT_GPIO_XEC_GPIO000_036_IRQ,
-		DT_GPIO_XEC_GPIO000_036_IRQ_PRIORITY,
-		gpio_gpio_xec_port_isr, DEVICE_GET(gpio_xec_port000_036), 0U);
+	IRQ_CONNECT(DT_INST_0_MICROCHIP_XEC_GPIO_IRQ_0,
+		DT_INST_0_MICROCHIP_XEC_GPIO_IRQ_0_PRIORITY,
+		gpio_gpio_xec_port_isr, DEVICE_GET(gpio_xec_port0), 0U);
 
-	irq_enable(DT_GPIO_XEC_GPIO000_036_IRQ);
+	irq_enable(DT_INST_0_MICROCHIP_XEC_GPIO_IRQ_0);
 #endif
 	return 0;
 }
-#endif /* CONFIG_GPIO_XEC_GPIO000_036 */
 
-#ifdef CONFIG_GPIO_XEC_GPIO040_076
-static int gpio_xec_port040_076_init(struct device *dev);
+DEVICE_AND_API_INIT(gpio_xec_port0, DT_INST_0_MICROCHIP_XEC_GPIO_LABEL,
+		gpio_xec_port0_init,
+		&gpio_xec_port0_data, &gpio_xec_port0_config,
+		POST_KERNEL, CONFIG_KERNEL_INIT_PRIORITY_DEFAULT,
+		&gpio_xec_driver_api);
+
+#endif /* DT_INST_0_MICROCHIP_XEC_GPIO */
```

Comment:
In the increasing number of the device drivers we use macros to define driver instances. This update may be a good opportunity to switch to this new style since there is a lot of repeated code. One example could be found in the `drivers/gpio/gpio_nrfx.c` driver.

Reply:
Unfortunately this is not possible since we have `#ifdef` conditions for some of the code. It would need to be repeated, or be constructed using templates similar to  `drivers/serial/uart_ns16550_port_x.h` which involves CMake doing the replacements. I thought it was simpler to leave it repeated for the time being. It might be follow up commit when the GIRQ aggregator is implemented properly and no `#ifdef` is needed to do the merging.

Reply:
But infact, the interrupt enable should depend on DT `IRQ` field, and DT GIRQ should go directly to GIRQ. So they are two separate things. I'll fixup this. This should also show why the earlier mentioned irq check makes sense.